### ISSUE TITLE
Cytoscape Expand Collapse plug-in

### ISSUE THREAD

**Issue by zhjuve** [rcarcasses/vue-cytoscape#8]
Hi,
Really great work done with component!
I am trying to use expand collapse plugin but i can only collapse parent nodes without edges, for nodes with outgoing edges i got total hang and high CPU usage.

**Comment by rcarcasses** [comment]
Hi @zhjuve, welcome! Can you try to provide a minimum reproducible example so I can reproduce your issue? At first sight it seems like it is mostly related with the plugin configuration/interaction rather than with `vue-cytoscape`.

**Comment by zhjuve** [comment]
Hi!
Here is very minimal example https://github.com/zhjuve/vuecytoaws
Context menu items have Collapse Expand commands, parent node VPC_default can be collapsed because there are no edges going/coming from outside, where e.g. VPC_Sweden cannot as there are "CONNECTS" edge from "0.0.0.0"

**Comment by zhjuve** [comment]
Once loaded press Shuffle layout to use some layout as json file have no included.

**Comment by rcarcasses** [comment]
Thanks for the example, gives a solid point for researching. Anyway I'll take a deeper look later, but from what I see the issues you are experiencing are related more to the behaviour of each of the `cytoscape` plugins themselves (or the combination of them probably) and not due to `vue-cytoscape`. Did you tried to have the same configuration with vanilla `cytoscape`? I would recommend you to do that, it can give you a hint of where the performance bottleneck actually happens.

**Comment by zhjuve** [comment]
Hi, thanks, actually i am implementing smth like https://github.com/duo-labs/cloudmapper, it is on pure Javascript and it is working there normally.
I found similar issue in https://github.com/cytoscape/cytoscape.js-cose-bilkent/issues/84 so my suspicion is that event handler cannot process change of edge end to collapsed parent node. But i am not a Vue specialist yet to try to disable event handling for cytoscape.

**Comment by rcarcasses** [comment]
This is good information, I will definitively take a look later and will let you know!

**Comment by rcarcasses** [comment]
Hey there, I just found that there is indeed some issue with the synchronisation between `vue-cytoscape` and `cytoscape`. In the mean time, this should help: update to the latest version and add these hooks to the `cytoscape` component

```
<template>
    <cytoscape
      :config="config"
      :preConfig="preConfig"
      :afterCreated="afterCreated"
      style="width: 1500px; height: 800px;"
      :debug="true"
      :elementAdded="elementAdded"
      :elementRemoved="elementRemoved"
    />
</template>

export default {
  methods: {
    elementAdded(el) {
      // do nothing by now
      console.log("element added", el);
    },
    elementRemoved(el) {
      // do nothing by now
      console.log("element removed", el);
    },
...
}
```

**Comment by rcarcasses** [comment]
Hi there, the issue you were experiencing has been fixed in version `v0.2.8`. I have tested it locally.

Please update to the latest version and let me know any other issue you may experience.

**Comment by zhjuve** [comment]
Workig now! Thanks a lot!
